scripts/utils.py

```diff
@@ -0,0 +1,9 @@
+def parse_recipe_id(recipe_id):
```

Comment:
I understand the rationale for wanting to put this function in a separate file, since both `linear_solver.py` and `flow_chart.py` depend on it. However I think overall I still prefer it in `linear_solver.py`, since it was defined next to its counterpart that sets up the `recipe_id` in the first place, which adds some context and IMO makes it easier to tell what it's doing. Would you be able to move it back, and import this function from `linear_solver` in the `flow_chart` module?

Reply:
it's not possible because circular imports. would need to refactor it some other way or use dependency injection could work like putting parse_recipe_id as a function of linear solver and pass the whole object to flow chart maker.